AddNewRepairView

vanyalaptii opened the pull request:
1. AddNewRepairView added.
2. Created logic for add new repairs to database.

vanyalaptii committed:
AddNewRepairView added

1. AddNewRepairView added.
2. Created logic for add new repairs to database.

vanyalaptii committed:
RepairsListView

RepairsListView logic.

OksanaFedorchuk commented on ReFix.xcodeproj/project.pbxproj:
```diff
@@ -178,13 +196,16 @@
 		C35B38062B514E2500D983D8 /* Modules */ = {
 			isa = PBXGroup;
 			children = (
+				C3B5EB7D2B630C97003524FA /* RepairsView */,
```
В тебе модулі мають в назві View, а лежать там і в"ю моделі і менеджери. Просто - Repairs
Auth
Profile

vanyalaptii replied:
Виправлено

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListView.swift:
```diff
@@ -0,0 +1,127 @@
+        .onAppear {
+            UIBarButtonItem.appearance(whenContainedInInstancesOf: [UISearchBar.self]).title = "Відмінити"
+            Task {
+                try await viewModel.loadCurrentUser()
```
Чому ти в в"ухах викликаєш завантаження даних? Нехай цим займається в"ю модель сама, нащо В2ю про це знати? йому треба тільки відображати далі які йому дасть ВМ
(запускай ці методи в ініті в"ю моделі)

vanyalaptii replied:
Виправлено

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListView.swift:
```diff
@@ -0,0 +1,127 @@
+    
+    @State private var searchText = ""
+    @State private var searchIsActive = false
+    @State var repairsListArray: [Repair] = []
```
Це ти взагалі ніде не юзаєш

vanyalaptii replied:
Прибрибрав у VM

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListView.swift:
```diff
@@ -0,0 +1,127 @@
+    
+    func updateList() {
+        withAnimation {
+            self.repairsListArray = viewModel.repairListArray
```
це зайве

vanyalaptii replied:
Прибрав

OksanaFedorchuk commented on ReFix/Modules/RepairsView/AddNewRepairView.swift:
```diff
@@ -0,0 +1,130 @@
+        case employee
+    }
+    
+    @EnvironmentObject private var viewModel: RepairsListViewModel
```
ти для цього View юзаєш чужий ViewModel
в результаті, в тебе ViewModel юзається в двох різних місцях, в яких різна ціль і засоби, і В"ю мають доступ до проперті і методів, які вони не юзають в своєму ЮІ (їм не потрібні)
Це порушення принципу I - Interface Segregation (SOLID) 
а також S - single responsibility (SOLID)

vanyalaptii replied:
Виправив. Зробив для AddNewRepairsView власний VM

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListView.swift:
```diff
@@ -0,0 +1,127 @@
+    
+    @StateObject var viewModel = RepairsListViewModel()
+    
+    @State private var searchText = ""
```
В тебе проперті в В"ю, хоча воно тісно пов"язані з логікою, і повинне бути в ВМ

vanyalaptii replied:
Виправив

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsListView.swift:
```diff
@@ -0,0 +1,127 @@
+    }
+    
+    var searchResults: [Repair] {
+        if searchText.isEmpty {
```
Ти здійснюєш фільтрацію в В"ю, хоча в тебе є в"ю модель для логіки

vanyalaptii replied:
Виправив

OksanaFedorchuk commented on ReFix/Modifiers/ClearButtonViewModifier.swift:
В тебе файл називається ClearButtonViewModifier - для кастомізації кнопок краще використовувати ButtonStyle

також, в цьому файлі треба поправити форматування. - 
struct ClearButtonViewModifier: ViewModifier
{
--->
struct ClearButtonViewModifier: ViewModifier {

і для чого тобі там public ?

vanyalaptii replied:
Якщо я вірно все зрозумів ButtonStyle змінює зовнішній вигляд кнопки, а мені потрібно помістити кнопку у кастомний текстфілд. Але якщо я не правий сподіваюсь буде можливість пояснити мені як це правильно зробити

OksanaFedorchuk commented on ReFix/Modules/RepairsView/RepairsManager.swift:
```diff
@@ -0,0 +1,54 @@
+    
+    static var shared = RepairsManager()
+    
+    private init() {}
```
для чого це?

vanyalaptii replied:
Прибрано

OksanaFedorchuk commented on ReFix/Modules/ProfileView/UserManager.swift:
```diff
@@ -0,0 +1,45 @@
+final class UserManager {
+    
+    static let shared = UserManager()
+    private init(){}
```
пуста функція

vanyalaptii replied:
Прибрано

OksanaFedorchuk commented on ReFix/Modules/ProfileView/UserManager.swift:
```diff
@@ -0,0 +1,45 @@
+    static let shared = UserManager()
+    private init(){}
+    
+    private let encoder: Firestore.Encoder = {
```
винеси приватне нижче, можна в extension
спочатку йде важливе і неприватне потім приватне

vanyalaptii replied:
Треба взагалі все приватне вниз прибрати? Чи всеж в першу чергу зверху проперті, потім методи а їх в же в свою чергу відсортувати по приватності?

vanyalaptii replied:
Бо я так і зробив... Спочатку протерті, потім методи і і в них вже все приватне опустив нижче. Будь ласка дай знати якщо треба просто все приватне вниз

OksanaFedorchuk replied:
зручно, коли методи які використовуються не тільки в цьому файлі, а і зовні, ідуть перші

OksanaFedorchuk replied:
це справа смаку, роби як хочеш

OksanaFedorchuk commented on ReFix/Modules/RepairsView/AddNewRepairView.swift:
```diff
@@ -0,0 +1,130 @@
+            .toolbar {
+                ToolbarItem {
+                    Button {
+                        viewModel.addNewRepairIsPresented = false
```
viewModel.isAddNewRepairPresented

vanyalaptii replied:
Виправив

OksanaFedorchuk commented on ReFix/Modules/ProfileView/ProfileView.swift:
```diff
@@ -84,7 +84,10 @@ struct ProfileView: View {
+            .navigationBarTitle(viewModel.user?.companyName ?? "Налаштування")
+        }
+        .task {
+            try? await viewModel.loadCurrentUser()
```
в ВМ

vanyalaptii replied:
Виправив

vanyalaptii committed:
Modules names changed

1. Modules names changed.
2. Logic moved to viewModels.
3. Removed unused function and properties.
4.AddNewRepairViewModel created.
